packages/neon/neon/lib/src/widgets/image.dart

```diff
@@ -328,13 +286,118 @@ class NeonApiImage extends StatelessWidget {
   }
 }
 
+/// Nextcloud URL image.
+///
+/// Extension for [NeonCachedImage] providing a [NextcloudClient] to the caller
+/// to retrieve the image from an URL.
+///
+/// See:
+///  * [NeonCachedImage] for a customized image
+///  * [NeonImageWrapper] for a wrapping widget for images
+class NeonUrlImage extends StatelessWidget {
+  /// Creates a new Neon URL image with the active account.
+  ///
+  /// See [NeonUrlImage.withAccount] for using a specific account.
+  const NeonUrlImage({
+    required this.url,
+    this.reviver,
+    this.writeCache,
+    this.isSvgHint = false,
+    this.size,
+    this.fit,
+    this.svgColorFilter,
+    this.errorBuilder,
+    super.key,
+  }) : account = null;
+
+  /// Creates a new Neon URL image with the given [account].
+  ///
+  /// See [NeonUrlImage] for using the active account.
+  const NeonUrlImage.withAccount({
+    required this.url,
+    required Account this.account,
+    this.reviver,
+    this.writeCache,
+    this.isSvgHint = false,
+    this.size,
+    this.fit,
+    this.svgColorFilter,
+    this.errorBuilder,
+    super.key,
+  });
+
+  /// Account to use for the request.
+  ///
+  /// Defaults to the currently active account in [AccountsBloc.activeAccount].
+  final Account? account;
+
+  /// Image url.
+  final String url;
+
+  /// Custom cache reviver function.
+  final CacheReviver? reviver;
+
+  /// Custom cache writer function.
+  final CacheWriter? writeCache;
+
+  /// {@macro NeonCachedImage.svgHint}
+  final bool isSvgHint;
+
+  /// {@macro NeonCachedImage.size}
+  final Size? size;
+
+  /// {@macro NeonCachedImage.fit}
+  final BoxFit? fit;
+
+  /// {@macro NeonCachedImage.svgColorFilter}
+  final ColorFilter? svgColorFilter;
+
+  /// {@macro NeonCachedImage.errorBuilder}
+  final ErrorWidgetBuilder? errorBuilder;
+
+  @override
+  Widget build(final BuildContext context) {
+    final account = this.account ?? NeonProvider.of<AccountsBloc>(context).activeAccount.value!;
+
+    return NeonCachedImage(
+      getImage: () async {
+        final uri = account.completeUri(Uri.parse(url));
+        final headers = <String, String>{};
+
+        if (uri.toString().startsWith(account.serverURL.toString()) && account.client.authentications.isNotEmpty) {
```

Comment:
Please add a comment that this is done to avoid leaking the authentication to other servers.
I feel like this is very important.